src/include/duckdb/storage/compression/alprd/algorithm/alprd.hpp

```diff
@@ -0,0 +1,221 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/storage/compression/alprd/algorithm/alprd.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/storage/compression/patas/patas.hpp"
+#include "duckdb/storage/compression/alprd/alprd_constants.hpp"
+
+#include "duckdb/common/bitpacking.hpp"
+#include "duckdb/common/common.hpp"
+#include "duckdb/common/pair.hpp"
+#include "duckdb/common/unordered_map.hpp"
+#include "duckdb/common/limits.hpp"
+
+namespace duckdb {
+
+namespace alp {
+
+template <class T, bool EMPTY>
+class AlpRDCompressionState {
+public:
+	using EXACT_TYPE = typename FloatingToExact<T>::type;
+
+	AlpRDCompressionState() : right_bw(0), left_bw(0), exceptions_count(0) {
+	}
+
+	void Reset() {
+		left_bp_size = 0;
+		right_bp_size = 0;
+		exceptions_count = 0;
+	}
+
+public:
+	uint8_t right_bw;
+	uint8_t left_bw;
+	uint16_t exceptions_count;
+	uint8_t right_parts_encoded[AlpRDConstants::ALP_VECTOR_SIZE * 8];
+	uint8_t left_parts_encoded[AlpRDConstants::ALP_VECTOR_SIZE * 8];
+	uint16_t left_parts_dict[AlpRDConstants::DICTIONARY_SIZE];
+	uint16_t exceptions[AlpRDConstants::ALP_VECTOR_SIZE];
+	uint16_t exceptions_positions[AlpRDConstants::ALP_VECTOR_SIZE];
+	idx_t left_bp_size;
```

Comment:
Same as above, not sure what this variable represents

Reply:
Yes. We have changed the variable names to be more representative